### PR TITLE
fix : Modified the response  of http exception and added mock stub

### LINKED ISSUE #1968: Unit Test Fails for LoginPresenterTest
**Summary:**

In the LoginPresenterTest ,  four of the unit test fails under that of method :  testLoadCLientFails and testLoadClientUnauthorized   

**Steps to reproduce:**

 Run the LoginPresenterTest class , we shall sell the failing result for above mentioned  unit tests .

**Expected behaviour:**

All of the  seven  unit test should pass 

**Observed behaviour:**

 Test under the testLoadCLientFails and testLoadClientUnauthorized fails  with error : " message = null "  and "unfinished stubbing " 

**Device and Android version:**

Ocurred while running the unit test in IDE

**Screenshots:**
image

### PR DESCRIPTION
**Pull request by NiranjanNlc**
Fixes #1968 and #1977

 
image

image

### CODE REVIEW

**Comment by NiranjanNlc** [comment]
hey @jawidMuhammadi  , would you please review this PR and suggest any changes ?

File: app/src/commonTest/java/org/mifos/mobile/RetrofitUtils.kt [hunk 1]
```diff
@@ -10,10 +11,21 @@ import retrofit2.Response
...
+        return HttpException(getResponseForError(404, "Not Found"))
     }
 
     fun get401Exception(): Exception {
-        return HttpException(Response.error<Any>(401, ResponseBody.create(MediaType.parse("application/json"), "UnAuthorized")))
+        return HttpException(getResponseForError(401, "Unauthorized"))
+    }
+
+    //generate response for error based on code and messaghe
+    private fun getResponseForError(errorCode: Int, message : String ): Response<Any> {
```
Comment (jawidMuhammadi) [inline app/src/commonTest/java/org/mifos/mobile/RetrofitUtils.kt]: The method name is self-explanatory enough, you could remove the comment.

File: app/src/commonTest/java/org/mifos/mobile/RetrofitUtils.kt [hunk 2]
```diff
@@ -10,10 +11,21 @@ import retrofit2.Response
  */
 object RetrofitUtils {
     fun get404Exception(): Exception {
-        return HttpException(Response.error<Any>(404, ResponseBody.create(MediaType.parse("application/json"), "Not Found")))
+        return HttpException(getResponseForError(404, "Not Found"))
     }
 
     fun get401Exception(): Exception {
```
Comment (jawidMuhammadi) [inline app/src/commonTest/java/org/mifos/mobile/RetrofitUtils.kt]: Remove this and the `get404Exception` method and directly use the `getResponseForError` method.

File: app/src/test/java/org/mifos/mobile/UpdatePasswordPresenterTest.kt [hunk 3]
```diff
@@ -54,6 +54,7 @@ class UpdatePasswordPresenterTest {
     fun setUp() {
         Mockito.`when`(preferencesHelper?.baseUrl)
                 .thenReturn(BaseURL.PROTOCOL_HTTPS + BaseURL.API_ENDPOINT)
+        Mockito.`when`(preferencesHelper?.userName).thenReturn("username") // username is required to update authentication token
```
Comment (jawidMuhammadi) [inline app/src/test/java/org/mifos/mobile/UpdatePasswordPresenterTest.kt]: Remove the comment, please

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

**Comment by NiranjanNlc** [comment]
hey @jawidMuhammadi , i have made the necessary changes

**Review by jawidMuhammadi — APPROVED** [review]
LGTM! 👏

**Pull request merged**